### DISCUSSION TITLE
How to rename a source  without re-ingestion of data?

### DISCUSSION THREAD

**Discussion by podshivalovdv** [vectordotdev/vector#14014] [Q&A] [answered]
Hi guys, 
I have a need to rename a file source  in Vector, because we need to split one source into 2. 

But if I do so  - vector will create new checkpoints and will read log files from the beginning. 
What is the best way to do that split without re-ingestion of all the data?

I see 1 option - stop vector, duplicate checkpoint folders using names of new sources and start vector again. Is that the right way? 

Thanks.

**Comment by jszwedko** [comment, marked as answer]
Hi @podshivalovdv !

What you suggested seems like the best approach to me: copying the checkpoint folder. That should allow each source to continue from where it left off for its respective files.

**Reply by podshivalovdv** [reply]
Great, thank you Jesse!
